src/binaryutils.cpp

```diff
@@ -0,0 +1,259 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "status.hpp"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wall"
+#include "tensorflow/core/framework/tensor.h"
+#include "tensorflow_serving/apis/prediction_service.grpc.pb.h"
+#pragma GCC diagnostic pop
+
+#include "binaryutils.hpp"
+#include "logging.hpp"
+#include "opencv2/opencv.hpp"
+
+#include <inference_engine.hpp>
+
+namespace ovms {
+
+int getMatTypeFromTensorPrecision(InferenceEngine::Precision tensorPrecision) {
+    switch (tensorPrecision) {
+    case InferenceEngine::Precision::FP32:
+        return CV_32F;
+    case InferenceEngine::Precision::FP16:
+        return CV_16F;
+    case InferenceEngine::Precision::I16:
+        return CV_16S;
+    case InferenceEngine::Precision::U8:
+        return CV_8U;
+    case InferenceEngine::Precision::I8:
+        return CV_8S;
+    case InferenceEngine::Precision::U16:
+        return CV_16U;
+    case InferenceEngine::Precision::I32:
+        return CV_32S;
+    default:
+        return -1;
+    }
+}
+
+bool isPrecisionEqual(int matPrecision, InferenceEngine::Precision tensorPrecision) {
+    int convertedTensorPrecision = getMatTypeFromTensorPrecision(tensorPrecision);
+    if (convertedTensorPrecision == matPrecision) {
+        return true;
+    }
+    return false;
+}
+
+template <typename T>
+std::vector<T> reorder_to_nchw(const T* nhwcVector, int rows, int cols, int channels) {
+    std::vector<T> nchwVector(rows * cols * channels);
+    for (int y = 0; y < rows; ++y) {
+        for (int x = 0; x < cols; ++x) {
+            for (int c = 0; c < channels; ++c) {
+                nchwVector[c * (rows * cols) + y * cols + x] = reinterpret_cast<const T*>(nhwcVector)[y * channels * cols + x * channels + c];
+            }
+        }
+    }
+    return std::move(nchwVector);
+}
+
+cv::Mat convertStringValToMat(const std::string& stringVal) {
+    std::vector<unsigned char> vectordata(stringVal.begin(), stringVal.end());
+    cv::Mat data_mat(vectordata, true);
+
+    return cv::imdecode(data_mat, cv::IMREAD_UNCHANGED);
+}
+
+Status convertPrecision(const cv::Mat& src, cv::Mat& dst, const InferenceEngine::Precision requestedPrecision) {
+    int type = getMatTypeFromTensorPrecision(requestedPrecision);
```

Comment:
Does the return type have any alias? Perhaps it's worth creating a type for that value if there's not?